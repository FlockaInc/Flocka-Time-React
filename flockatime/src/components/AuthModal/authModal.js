import React, { useState, useEffect } from 'react';
import Modal from 'react-modal';

import ns from '../../utilities/notificationService';

const customStyles = {
  content: {
    top: '50%',
    left: '50%',
    right: 'auto',
    bottom: 'auto',
    marginRight: '-50%',
    transform: 'translate(-50%, -50%)',
    maxWidth: '450px',
    minWidth: '350px',
    padding: 0
  }
};

function AuthModal() {
  const [modalIsOpen, setModalIsOpen] = useState(false);

  useEffect(() => {
    ns.addObserver('AUTH_MODAL_TOGGLE', this, toggleModal);
    Modal.setAppElement('body');

    return () => {
      ns.removeObserver(this, 'AUTH_MODAL_TOGGLE')
    }
  });

  function toggleModal() {
    setModalIsOpen(!modalIsOpen);
  }

  return (
    <div className="modal fade" role="dialog">
      <div className="modal-dialog" role="document">
        <Modal
          isOpen={modalIsOpen}
          onRequestClose={toggleModal}
          shouldCloseOnOverlayClick={true}
          style={customStyles}
        >
          <div className="modal-content">
            <div className="modal-header">
              <h5 className="modal-title text-black">Modal title</h5>
              <button type="button" className="close" data-dismiss="modal" aria-label="Close">
                <span aria-hidden="true">&times;</span>
              </button>
            </div>
            <div className="modal-body">
              <p>Modal body text goes here.</p>
            </div>
            <div className="modal-footer">
              <button type="button" className="btn btn-primary">Save changes</button>
              <button type="button" className="btn btn-secondary" data-dismiss="modal">Close</button>
            </div>
          </div>
        </Modal>
      </div>
    </div>
  );
}

export default AuthModal;