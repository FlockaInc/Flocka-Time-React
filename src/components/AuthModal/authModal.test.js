import React from 'react';
import { render, cleanup } from '@testing-library/react';
import AuthModal from './authModal';

afterEach(cleanup);

it('renders', () => {
  const { asFragment } = render(<AuthModal />);
  expect(asFragment()).toMatchSnapshot();
});