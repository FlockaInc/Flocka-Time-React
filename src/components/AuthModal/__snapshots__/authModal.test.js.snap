// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="modal fade"
    id="sign-in-form"
  >
    <div
      class="modal-dialog modal-dialog-centered"
    />
  </div>
</DocumentFragment>
`;
